src/main/java/ejercicio/banco/repository/CsvAccountRepository.java

```diff
@@ -56,58 +59,46 @@ public CsvAccountRepository(){
 
     @Override
     public Account find(int id) throws FileNotFoundException {
-        //Since we dont want to give the method a file parameter, Im hardcoding the filename
-        String filename = String.join(File.separator, "src", "main", "resources", "csv", "accounts.csv");
-        List<String> in = reader.read(filename);
-
-        String line = in.get(id);
-        System.out.println(line);
-
-        // Assuming that the info is separated by "; " ,splits it
-        String[] parts = line.split(";");
-        String userName = parts[0];
-        float money = Float.valueOf(parts[1]);
-        String iban = parts[2];
+        List<Account> all = findAll(FILENAME);
 
-        //Prints user into console
-        System.out.println("User " + id + " named " + userName + " has been processed");
-
-        // Creates a new account and returns it
-        Account acc = new Account(id, userName, money, iban);
-        return acc;
+        for (Account account : all) {
+            if (id == account.getId()) {
+                return account;
+            }
+        }
+        //Method requires a return statement
+        return all.get(0);
```

Comment:
I was waiting how you solved this situation :)

We will introduce here an exception. We will discuss after how to do it, but in the mean time, just `return null`.

Reply:
In fact, I'm thinking that you will throw an exception here now. If you don't know how to do it, just let me know :)